data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/EventMatcher.java

```diff
@@ -149,4 +150,21 @@ private static boolean isNotEmpty(final String value) {
       .name()
       .toLowerCase();
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    EventMatcher that = (EventMatcher) o;
+    return Objects.equals(attributes, that.attributes);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(attributes);
+  }
```

Comment:
I think you don't want this since the type of `attributes` is `List<Entry<Function<CloudEvent, String>, String>>`, with the suggestion above we don't need these overrides.

Reply:
ah ah!